kratos/sources/model_part_io.cpp

```diff
@@ -504,6 +504,245 @@ void ModelPartIO::WriteConditions(ConditionsContainerType const& rThisConditions
+            if (check_same_type_vector[i] != rCheckSameTypeVectorPrevious[i]) return false;
+        }
+        return true;
+    };
+
+    // If there are master slave constraints we print them
+    if (rMasterSlaveConstraintContainer.size() > 0) {
+        // Define the name of the master slave constraint
+        std::string master_slave_constraint_name;
+
+        // Define the variables names
+        std::vector<std::string> variables_names;
+
+        // Define empty process info
+        ProcessInfo current_process_info;
+
+        auto it_master_slave_constraint_begin = rMasterSlaveConstraintContainer.begin();
+        auto master_slave_constraints_components = KratosComponents<MasterSlaveConstraint>::GetComponents();
+
+        // First we do the first master_slave_constraint
+        CompareElementsAndConditionsUtility::GetRegisteredName(*it_master_slave_constraint_begin, master_slave_constraint_name);
+
+        // We get the transformation matrix and the constant vector
+        Matrix transformation_matrix;
+        Vector constant_vector;
+
+        // Define the dofs
+        MasterSlaveConstraint::DofPointerVectorType master_dofs, slave_dofs;
+
+        // Compute the dofs
+        it_master_slave_constraint_begin->GetDofList(slave_dofs, master_dofs, current_process_info);
+
+        // We get the number of master and slave dofs
+        SizeType number_of_master_dofs = master_dofs.size();
+        SizeType number_of_slave_dofs = slave_dofs.size();
+        variables_names.reserve(number_of_master_dofs + number_of_slave_dofs);
+        check_same_type_vector_previous.reserve(2 + number_of_master_dofs + number_of_slave_dofs);
+        check_same_type_vector_previous.push_back(number_of_master_dofs);
+        check_same_type_vector_previous.push_back(number_of_slave_dofs);
+
+        for (IndexType i = 0; i < number_of_master_dofs; ++i) {
+            const auto& p_dof = master_dofs[i];
+            const auto& r_variable = p_dof->GetVariable();
+            variables_names.push_back(r_variable.Name());
+            check_same_type_vector_previous.push_back(r_variable.Key());
+        }
+        for (IndexType i = 0; i < number_of_slave_dofs; ++i) {
+            const auto& p_dof = slave_dofs[i];
+            const auto& r_variable = p_dof->GetVariable();
+            variables_names.push_back(r_variable.Name());
+            check_same_type_vector_previous.push_back(r_variable.Key());
+        }
+
+        // We get the transformation matrix and the constant vector
+        it_master_slave_constraint_begin->CalculateLocalSystem(transformation_matrix, constant_vector, current_process_info);
+
+        (*mpStream) << "Begin MasterSlaveConstraints\t" << master_slave_constraint_name << "\t" << number_of_master_dofs << "\t" << number_of_slave_dofs;
+        for (IndexType i = 0; i < variables_names.size(); ++i) {
+            (*mpStream) << "\t" << variables_names[i];
+        }
+        (*mpStream) << std::endl;
```

Comment:
`std::endl` is unnecessary and makes writing **very** expensive. Replace them with `\n` please.

Reply:
You are right